### ISSUE TITLE
Reviews Meeting -- March 15th

### ISSUE THREAD

**Issue by ababaian** [bioSyntax/bioSyntax#19]
Meeting to discuss the main review points, rebuttals and moving forward.

Date: Thursday March 15th
Time: 5:30 pm
Place: BCCRC

### Development

- Development for Pico
- Other Dev issues 

### Mozilla Sprint -- May 10 - 11 (Alyssa Lead)
- Update / Info
- [Project Link](https://github.com/mozilla/global-sprint/issues/154)
- [ TODO List ](https://github.com/bioSyntax/bioSyntax/issues/2) / Define goals?

### Actionable Reviewer Comments
We got our reviewer comments back, ultimately the whole critique can be summarized into two points which we need to provide more evidence for.
- [ ] Syntax highlighting intrinsically improves reading / comprehension of data versus uncolored text.
- [ ] There is a broad user base for bioSyntax, and it is becoming adopted.

> " In this manuscript there is no experiment performed or evidence presented or work referenced that shows highlighted text improves readability and understandability of these files."

- [ ]  We cite the research on syntax highlighting that show it does improve user performance, this wasn't clear in the text if the reviewer missed it.
- [ ] Color vision has evolved to resolve patterns, we can easily argue that the reason we have color is precisely to be able to see patterns in an otherwise uncolored world. {do literature review on this}
- [ ] There should be some basic psychology experiments which test the effect color has in human text/number perception {find + cite). Like for Synaesthesia

 ![example](https://media.mnn.com/assets/images/2014/09/800px-Synaesthesiatest.jpg)

"As you can see, the process of finding the 2�s within the field of 5�s becomes much easier when each digit is associated with a different color. Though the perceived color of the numbers varies between synesthetes, the ability of someone with synesthesia to process visual search fields such as Figure 1 more quickly than the average person remains consistent across all cases. "
From:  [Here (to start lit review)](https://www.macalester.edu/projects/UBNRP/gcsynesthesia/introduction.html)

> "Testing whether seemingly trivial changes improves understanding or usability is a practice known as User Experience (UX) on which there is actually a (slightly informal) body of work around testing whether features such as this actually improve what the user understands and how quickly (https://en.wikipedia.org/wiki/User_experience_design#Testing_the_design)."

- [ ] Design a UX survey
- [ ] What are the key questions we would want to ask users?
- [ ] Experimental testing of user comprehension performance (how long to identify a SNP in a sequence etc...).
- [ ] Ethics approval on testing human subjects. --> Perform informal UX for us as designers and post results on website and refer to those results in manuscript as supplemental information. This way we are not running a formal human study.

> "In fact syntax highlighting is sufficiently easy to implement that any competent bioinformatician who wanted this could do so in an hour or so without bioSyntax (it takes almost as long to install). That most don't bother speaks volumes about how useful it may really be."

lol, have fun with VCF/SAM. So this is an opinion and what we need is evidence. 
- [ ] Demonstrate/quantify the user base for bioSyntax (release v0.1, widely publicize).
- [ ] Demonstrate/quantify that once made aware that such an improvement exists, users will accept and want exactly such an implement. This probably can fall under UX survey.

> "In fact syntax highlighting is sufficiently easy to implement that any competent bioinformatician who wanted this could do so in an hour or so without bioSyntax (it takes almost as long to install)"

- [ ] We need to streamline the installer process. This is easily going to be one of the biggest hurdles to getting users on-board and having a postive user experience.

> "My only issue is that I would disagree with the authors on one comment: "However, these applications require extra user time to learn and competently operate" for tools such as IGV and JalView. These tools are very user friendly, and would argue that something like Vi/Vim is far harder to operate. However, I do agree that text editors and pagers are vastly quicker, whilst not as feature rich, as desktop genomic feature viewers. This doesn't affect my recommendation for acceptance, but the authors might wish to clarify this statement."

Totally fair, we should reword this section to emphasize that bioSyntax is a supplement for text-editor/pager based data analysis and never intended to replace full-blown graphical user interfaces. Our wording here may have come across a bit strong.

> "I commend the authors for following open science principles and releasing the software for community contributions on GitHub, and making this manuscript available via bioRxiv prior to publication. I see from the GitHub issues that users are already starting to contribute code to incorporate bioSyntax into other editors, which is great. This is a very nice, short paper on what looks to be a helpful piece of software."

<3

### 

- ...

**Comment by lazypanda10117** [comment]
Hi Artem, can I join remotely for this meeting?

**Comment by ababaian** [comment]
Of course, we can try skype and/or discord in a pinch. Maybe we call you in at 5:20 then so we can start promptly at 5:30.

**Comment by ababaian** [comment]
@lazypanda10117 What's your skype ID?

**Comment by lazypanda10117** [comment]
jeffreykam-0415, thanks!

**Comment by Jwong684** [comment]
Simple survey to show usability:
Survey questions:
1) Which text editors/pagers do you use for reading biological formats?
>check-boxes + other
2) What biological files do you look at commonly (in the last 12 months)?
>check-boxes + other
3) What other editors or biological files would you like to see in the future?
>comments
4) How difficult was it to install bioSyntax?
1-10
5) Which part was hard to install?
>comments
6) Do you use syntax highlighting in any context?
1-10
7) What area would you like to see improvement in?
>comments
8) picture with/without bioSyntax colouring: Which picture more clearly conveys the content?

9) picture with sam file with/without bioSyntax colouring: Which read has the highest density of CpGs?
picture with mfa file with/without bioSyntax colouring: Find and copy the read that has a SNP.
>need timed responses (need a platform that supports timing)

**Comment by Jwong684** [comment]
Hey Jeff, I can't find you on skype. Add me instead: jasperwong_123@hotmail.com

**Comment by lazypanda10117** [comment]
Hi Jasper, I think I have added you on Skype now?

**Comment by alyeffy** [comment]
Questions to add to the survey (for Mozilla Sprint development):
1. If applicable, which text editors/IDEs do you use and are there any that you would like to be compatible with bioSyntax?
2. What are some features that you think would be useful for your work (or specific file formats) if bioSyntax incorporated them?
3. If any, what other file formats (including non-biological) would you like to have support for?

**Comment by Jwong684** [comment]
Hi all,

I have tried a number of survey tools, and it seems that the only ones that have timers per question are the ones that require a subscription fee. (I cannot find a free online tool that lets you do so; if you know of any, let me know). The best one I have found so far is survey crest, though the timer is based on the whole quiz instead of per question. I adapted to this by generating this general survey, with links at the bottom of the survey linking to two independent tests (one with biosyntax on and one with biosyntax off); both of which has a timer of 5 minutes.

https://www.surveycrest.com/s/5ab931f41e4ba

Try it out as a test run.

Jasper

**Comment by ababaian** [comment]
Hey @Jwong684 I think that survey is too complicated and all the BW and all the color images are segregated. @fransilvion said he'll look into it as well.

I love the graphics and the questions though, it took me quite a while to figure them out in BW. Definitely proved it to myself.

**Comment by alyeffy** [comment]
I think the graphics and questions are great too. It might be helpful to ask the user how helpful they think syntax highlighting will be for their work before they do the test, and ask them that question again after they do the test.

As for timed surveys, maybe check to see if Google Forms have an add-on for timed questions that you could use, or timed pages and put the test questions on a different page of the form.

**Comment by Jwong684** [comment]
Thanks for giving it a trial run guys. I realized that the data file that SurveyCrest gives doesn't collect the time for the surveys, so I would need to manually scroll through every response for the times. Are there any questions on the survey you wanted to modify/add/remove? I think a max limit of 10 questions is a good aim.

The main limiting factor at the moment is finding a free survey platform that gives a timer/question (most platforms require a large subscription fee for that service) and also a way to collect all these times in a data form. If we could think of a way to show the efficacy of bioSyntax without using timers, that would make things a lot easier.

@alyeffy I've checked Google Forms; they have an add-on called Timify, but that too is a timer for the length of the survey and not per question/section.
